### PR TITLE
feat: expose fetch to custom functions

### PR DESCRIPTION
**Pull request by P0lip**
**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

Make sure to select this option before you start reviewing image
I moved a few tests under the same block and therefore the diff is slightly noisier than it could be.

I didn't update documentation, since despite having access to fetch, you cannot really use it until the support for async functions #1058 is brought.

### CODE REVIEW

**Comment by P0lip** [comment]
@marbemac can you think of any functions that might be useful from the perspective of Studio user?

File: src/__tests__/linter.karma.test.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    spectral = new Spectral();
+  });
+
+  afterEach(() => {
+    window.fetch = fetch;
+  });
+
+  describe('custom functions', () => {
+    it('should be able to make a request using fetch', async () => {
+      fetchMock.mock('https://stoplight.io', {
```
Comment (nulltoken) [inline src/__tests__/linter.karma.test.ts:20]: @P0lip Just off-topic thought: This current test set up compels us to duplicate tests _and_ use two different libs to intercept fetch calls _and_, depending on the the scope, use 3 different suffixes for test files (jest, karma or void). 

This makes contributing and reviewing a little bit harder than it should be.

I feel there might be a better way to do this.

Thoughts?

Comment (P0lip) [reply src/__tests__/linter.karma.test.ts:20]: > @P0lip Just off-topic thought: This current test set up compels us to duplicate tests and use two different libs to intercept fetch calls and, depending on the the scope, use 3 different suffixes for test files (jest, karma or void).

Yeah, that is true.

> I feel there might be a better way to do this.

We could simply drop nock and use fetch-mock everywhere, I believe.
Nock is pretty cool though as it's an actual server.

Comment (nulltoken) [reply src/__tests__/linter.karma.test.ts:20]: @P0lip I'm going to log an issue about this to keep track of it.

**Comment by P0lip** [comment]
@nulltoken mind taking a stab at it?

**Pull request merged**